Libraries/Cordio/wsf/sources/targets/freertos/wsf_nvm.c

```diff
@@ -232,7 +287,11 @@ bool_t WsfNvmWriteData(uint64_t id, const uint8_t *pData, uint16_t len, WsfNvmCo
   }
 
   WSF_ASSERT(!((id == WSF_NVM_RESERVED_FILECODE) || (id == WSF_NVM_UNUSED_FILECODE)));
-  WSF_ASSERT((wsfNvmCb.availAddr - WSF_NVM_START_ADDR) <= wsfNvmCb.totalSize);
+
+  if (!wsfNvmHaveEnoughSpace(len)) {
```

Comment:
Should this be checking for `wsfNvmHaveEnoughSpace(len+sizeof(header))`? I am still seeing a HardFault, even with the additional fixes from today (which look great to me). I'm trying to figure out if it's a problem with my test setup or not.

Reply:
@EricB-ADI ^^^